### ISSUE TITLE
Silence the "org-ql: No headings in buffer: %s" ?

### ISSUE THREAD

**Issue by unhammer** [alphapapa/org-ql#175]
This is a minor thing, but the "org-ql: No headings in buffer: %s" message on empty buffers is a bit annoying when using ql non-interactively. (I have an agenda buffer filled by an ical subscription, it's often empty …) Could that message perhaps only be sent if called interactively, or perhaps there could be some variable we can set? Or should we just inhibit-message when calling org-ql (feels a bit heavy-handed)?

**Comment by alphapapa** [comment]
Please see https://github.com/alphapapa/org-ql/issues/28 and the associated commit.  I added the message to err on the side of caution, so users who search an empty buffer by accident will not mistakenly think that there are no results for their query.

If you want to add a variable to control whether the message is displayed, that would seem reasonable.

**Comment by colonelpanic8** [comment]
@alphapapa I don't think it makes any sense at all in the context of e.g. `org-ql-find-in-agenda`.

I, for example, have a inbox.org, which I use to quickly jot things down without trying to think about where they really belong. This file is actually often empty, and that is perfectly fine, but I obviously also want it to be included in my agenda files.

I'm assuming that `org-ql--select` is simply applied to each agenda file by `org-ql-find-in-agenda`, but I really don't see how it could be surprising to a user that no results come up if all of their agenda files happen to be empty, so I'm not sure what this message accomplishes in this case, especially given that they will see the message even if just one of their files is empty.

> If you want to add a variable to control whether the message is displayed, that would seem reasonable.

This seems like a reasonable approach, I suppose (although I'm sort of morally against most usage of `message`), as long as this is also the default set from `org-ql-find-in-agenda`

**Comment by alphapapa** [comment]
@colonelpanic8 I can see how your use case would make the message undesirable.  As I said on your PR, I think it would be best to add an option that users like you can set to disable the message.

**Comment by maikol-solis** [comment]
One annoying behavior in emacs-plus (and possibly other Emacs versions on macOS) is that every time this message appears, the frame popups even if it's minimized.

**Comment by SpenceSellers** [comment]
I would find this setting valuable. I have hundreds of documents, some too small to need headings, and get this warning when I use `org-ql-refile`. I'm fine with simply not refiling to the files without headings since they wouldn't match anything anyway.
